### ISSUE TITLE
Downloading Episodes behind HTTP Proxy fail with "HTTP Error 500: handshakefailed"

### ISSUE THREAD

**Issue by XceptN** [gpodder/gpodder#881]
On my laptop (Oracle Linux 7.8) I am required to run behind an HTTP/HTTPS proxy. I have env variables http_proxy and https_proxy set and exported.

I just installed gPodder from GitHub and imported my OPML file. The subscriptions and episodes were discovered perfectly.

I am using Python 3.6.8.

The "Stream" option for every episode is greyed out and when I try to download any episode, it directly fails with "Failed: HTTP Error 500: handshakefailed"

The terminal that I ran gpodder from provides details as follows: (excerpt)

```
Traceback (most recent call last):
  File "/home/oyuksel/app/gpodder/src/gpodder/download.py", line 863, in run
    headers, real_url = downloader.retrieve_resume(self.tempname, self.status_updated)
  File "/home/oyuksel/app/gpodder/src/gpodder/download.py", line 403, in retrieve_resume
    tempname, reporthook=reporthook)
  File "/home/oyuksel/app/gpodder/src/gpodder/download.py", line 315, in retrieve_resume
    fp = self.open(url, data)
  File "/usr/lib64/python3.6/urllib/request.py", line 1757, in open
    return getattr(self, name)(url)
  File "/usr/lib64/python3.6/urllib/request.py", line 1936, in open_http
    return self._open_generic_http(http.client.HTTPConnection, url, data)
  File "/usr/lib64/python3.6/urllib/request.py", line 1932, in _open_generic_http
    response.status, response.reason, response.msg, data)
  File "/home/oyuksel/app/gpodder/src/gpodder/download.py", line 246, in http_error
    return super().http_error(url, fp, errcode, errmsg, headers, data=data)
  File "/usr/lib64/python3.6/urllib/request.py", line 1952, in http_error
    return self.http_error_default(url, fp, errcode, errmsg, headers)
  File "/home/oyuksel/app/gpodder/src/gpodder/download.py", line 259, in http_error_default
    raise gPodderDownloadHTTPError(url, errcode, errmsg)
gpodder.download.gPodderDownloadHTTPError: ('//media.blubrry.com/security_all_in/api.podcache.net/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3', 500, 'handshakefailed')
1601642993.965950 [gpodder.download] ERROR: HTTP 500 while downloading "Sleepwalkers at CES": handshakefailed
```
This happens for every single podcast (out of 11 of them)...

Note that I can use wget with proxy and full URL to download the mp3. e.g.:

```
$ wget http://media.blubrry.com/security_all_in/api.podcache.net/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3
--2020-10-02 15:57:26--  http://media.blubrry.com/security_all_in/api.podcache.net/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3
Resolving <<proxy information deleted for privacy/security>>
Connecting to<<proxy information deleted for privacy/security>>... connected.
Proxy request sent, awaiting response... 302 Found
Location: http://api.podcache.net/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3 [following]
--2020-10-02 15:57:26--  http://api.podcache.net/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3
Reusing existing connection to <<proxy information deleted for privacy/security>>.
Proxy request sent, awaiting response... 301 Moved Permanently
Location: https://api.podcache.net:443/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3 [following]
--2020-10-02 15:57:27--  https://api.podcache.net/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3
Connecting to <<proxy information deleted for privacy/security>>... connected.
Proxy request sent, awaiting response... 302 Moved Temporarily
Location: https://stream.redcircle.com:443/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3 [following]
--2020-10-02 15:57:27--  https://stream.redcircle.com/episodes/626285ff-5574-4240-a2b3-7da95d87d08d/stream.mp3
Connecting to <<proxy information deleted for privacy/security>>... connected.
Proxy request sent, awaiting response... 200 OK
Length: 41558935 (40M) [audio/mpeg]
Saving to: ‘stream.mp3’

10% [==>                                    ] 4.189.779    780KB/s  eta 65s    ^C

```
I failed to pinpoint a specific reference to this problem... Any help is very much appreciated.

**Comment by XceptN** [comment]
New information: I have just went non-proxy for a quick while. The "Stream" option for episodes were "disabled" still, but download worked.

I need to get this working behind the company proxy.

**Comment by auouymous** [comment]
The podcast's site could be having problems since 500 means internal server error. Please provide a feed URL.

**Comment by auouymous** [comment]
Are you able to test current git or https://ci.appveyor.com/project/elelay/gpodder/builds/36488586/artifacts to see if this issue has been fixed by #843? I suspect it is a duplicate of #905 which was fixed by #843.

**Comment by XceptN** [comment]
Lost in time. Fill file new if needed.
